sync/http/handler.go

```diff
@@ -123,20 +124,14 @@ func handleSuccess(w http.ResponseWriter, r *http.Request, rsp *sync.Response, e
 }
 
 func handleError(w http.ResponseWriter, err error) {
-	switch err.(type) {
-	case *sync.ValidationError:
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
-	case *sync.UnauthorizedError:
-		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
-	case *sync.ForbiddenError:
-		http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
-	case *sync.NotFoundError:
-		http.Error(w, "", http.StatusNotFound)
-	case *sync.ServiceUnavailableError:
-		http.Error(w, http.StatusText(http.StatusServiceUnavailable), http.StatusServiceUnavailable)
-	default:
-		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+	// Assert error to type CustomError in order to leverage the code and payload values that such errors contain.
+	if err, ok := err.(*CustomError); ok {
+		w.Header().Set("X-Content-Type-Options", "nosniff")
+		w.WriteHeader(err.Code())
+		fmt.Fprintln(w, err.Payload())
```

Comment:
since the request has Accept and or Content type headers we should leverage the encoders here. check the function that calls handleError and pass in the encoder function that has been resolved from the headers.